core/Utils.vala

```diff
@@ -76,11 +76,49 @@ namespace Maya.Util {
     }
 
     /**
-     * Converts the given TimeType to a DateTime.
+     * Gets the timezone of the given TimeType as a GLib.TimeZone.
      */
     private TimeZone timezone_from_ical (ICal.Time date) {
+        // Special case: return default UTC time zone for all-day events
+        if (date.is_date ()) {
+            debug ("Given date is 'DATE' type, not 'DATE_TIME': Using timezone UTC");
+            return new TimeZone.utc ();
```

Comment:
This form of constructor does not appear in libical.vapi in the source code on on https://valadoc.org/libical-glib/ICal.Timezone.html yet the code compiles so I am not sure what is happening here.